src/Compatibility/ApiCompat/Microsoft.DotNet.ApiCompatibility/Runner/ApiCompatRunner.cs

```diff
@@ -13,44 +13,32 @@ namespace Microsoft.DotNet.ApiCompatibility.Runner
     /// <summary>
     /// Enqueues work items and performs api compatibility checks on them.
     /// </summary>
-    public class ApiCompatRunner : IApiCompatRunner
+    public class ApiCompatRunner(ISuppressibleLog log,
+        ISuppressionEngine suppressionEngine,
+        IApiComparerFactory apiComparerFactory,
+        IAssemblySymbolLoaderFactory assemblySymbolLoaderFactory) : IApiCompatRunner
     {
-        private readonly HashSet<ApiCompatRunnerWorkItem> _workItems = new();
-        private readonly ISuppressibleLog _log;
-        private readonly ISuppressionEngine _suppressionEngine;
-        private readonly IApiComparerFactory _apiComparerFactory;
-        private readonly IAssemblySymbolLoaderFactory _assemblySymbolLoaderFactory;
+        private readonly HashSet<ApiCompatRunnerWorkItem> _workItems = [];
 
         /// <inheritdoc />
         public IReadOnlyCollection<ApiCompatRunnerWorkItem> WorkItems => _workItems;
 
-        public ApiCompatRunner(ISuppressibleLog log,
-            ISuppressionEngine suppressionEngine,
-            IApiComparerFactory apiComparerFactory,
-            IAssemblySymbolLoaderFactory assemblySymbolLoaderFactory)
-        {
-            _suppressionEngine = suppressionEngine;
-            _log = log;
-            _apiComparerFactory = apiComparerFactory;
-            _assemblySymbolLoaderFactory = assemblySymbolLoaderFactory;
-        }
-
         /// <inheritdoc />
         public void ExecuteWorkItems()
         {
-            _log.LogMessage(MessageImportance.Low,
+            log.LogMessage(MessageImportance.Low,
                 string.Format(Resources.ApiCompatRunnerExecutingWorkItems,
                     _workItems.Count));
 
             foreach (ApiCompatRunnerWorkItem workItem in _workItems)
             {
-                IReadOnlyList<ElementContainer<IAssemblySymbol>> leftContainerList = CreateAssemblySymbols(workItem.Left, workItem.Options, out bool resolvedExternallyProvidedAssemblyReferences);
+                List<ElementContainer<IAssemblySymbol>> leftContainerList = CreateAssemblySymbols(workItem.Left, out bool resolvedExternallyProvidedAssemblyReferences);
```

Comment:
Why did you change them to List?

Reply:
An analyzer suggested that: prefer concrete return type